src/AutoRest.CSharp/LowLevel/Generation/LowLevelConvenienceMethodWriter.cs

```diff
@@ -1,119 +0,0 @@
-            writer.Line($"private static {typeof(RequestContext)} {defaultRequestContextName} = new {typeof(RequestContext)}();");
-
-            var methodSignature = new MethodSignature("FromCancellationToken", null, null, MethodSignatureModifiers.Internal | MethodSignatureModifiers.Static, typeof(RequestContext), null, new List<Parameter> { KnownParameters.CancellationTokenParameter });
-            using (writer.WriteMethodDeclaration(methodSignature))
-            {
-                using (writer.Scope($"if ({KnownParameters.CancellationTokenParameter.Name} == {typeof(CancellationToken)}.None)"))
-                {
-                    writer.Line($"return {defaultRequestContextName};");
-                }
-
-                writer.Line().Line($"return new {typeof(RequestContext)}() {{ CancellationToken = {KnownParameters.CancellationTokenParameter.Name} }};");
-            }
-            writer.Line();
-        }
-
-        private void WriteClientConvenienceMethod(string diagnosticsPropertyName, bool async)
-        {
-            using (WriteConvenienceMethodDeclaration(async))
-            {
-                if (_convenienceMethod.Diagnostic != null)
-                {
-                    using (WriteDiagnosticScope(_writer, _convenienceMethod.Diagnostic, diagnosticsPropertyName))
-                    {
-                        WriteClientConvenienceMethodBody(async);
-                    }
-                }
-                else
-                {
-                    WriteClientConvenienceMethodBody(async);
-                }
-            }
-
-            _writer.Line();
-        }
-
-        private CodeWriter.CodeWriterScope WriteConvenienceMethodDeclaration(bool async)
-        {
-            var methodSignature = _convenienceMethod.Signature.WithAsync(async);
-            _writer
-                .WriteMethodDocumentation(methodSignature)
-                .WriteXmlDocumentation("remarks", $"{methodSignature.DescriptionText}");
-            var scope = _writer.WriteMethodDeclaration(methodSignature);
-            if (_convenienceMethod.BodyParameter != null)
-            {
-                _writer.WriteParametersValidation(new[] { _convenienceMethod.BodyParameter });
-            }
-            return scope;
-        }
-
-        private void WriteClientConvenienceMethodBody(bool async)
-        {
-            string contextVariableName = _convenienceMethod.Signature.Parameters.FirstOrDefault(parameter => parameter.Name == KnownParameters.RequestContext.Name) != null ? $"{KnownParameters.RequestContext.Name}1" : KnownParameters.RequestContext.Name;
-            _writer.Line($"{typeof(RequestContext)} {contextVariableName} = FromCancellationToken({KnownParameters.CancellationTokenParameter.Name});");
-
-            string responseVariableName = _convenienceMethod.Signature.Parameters.FirstOrDefault(parameter => parameter.Name == "response") != null ? "response1" : "response";
```

Comment:
@pshao25 Variables should always be declared via `CodeWriterDeclaration`, so that `CodeWriter` can handle name conflicts